packages/lib-react-components/src/Markdownz/Markdownz.spec.js

```diff
@@ -11,4 +13,257 @@ describe('<Markdownz />', function () {
+      const imagePropsMock = { src: 'https://via.placeholder.com/350x350', alt: altText, children: undefined }
+      const imagePropsMockWithSize = { src: 'https://via.placeholder.com/350x350', alt: `${altText} =100x100`, children: undefined }
+      it('should return a Grommet Image component', function () {
+        wrapper.instance().renderMedia(imagePropsMock)
+        const returnedValue = renderMediaSpy.returnValues[0]
+        expect(returnedValue.type).to.equal(Image)
+      })
+
+      it('should set the alt attribute with the alt prop', function () {
+        wrapper.instance().renderMedia(imagePropsMock)
+        const returnedValue = renderMediaSpy.returnValues[0]
+        expect(returnedValue.props.alt).to.equal(imagePropsMock.alt)
+      })
+
+      it('should set the src attribute with the src prop', function () {
+        wrapper.instance().renderMedia(imagePropsMock)
+        const returnedValue = renderMediaSpy.returnValues[0]
+        expect(returnedValue.props.src).to.equal(imagePropsMock.src)
+      })
+
+      it('should use the width and height from the alt text if defined', function () {
+        wrapper.instance().renderMedia(imagePropsMockWithSize)
+        const returnedValue = renderMediaSpy.returnValues[0]
+        expect(returnedValue.props.width).to.equal('100')
+        expect(returnedValue.props.height).to.equal('100')
+      })
+
+      it('should remove the width and height declaration from the alt text before setting it on the rendered Image', function () {
+        wrapper.instance().renderMedia(imagePropsMockWithSize)
+        const returnedValue = renderMediaSpy.returnValues[0]
+        expect(returnedValue.props.alt).to.equal(altText)
+      })
+    })
+
+    describe('when the media is a video', function () {
+      const videoPropsMock = { src: 'https://static.zooniverse.org/www.zooniverse.org/assets/home-video.mp4', alt: 'Video of the Zooniverse', children: undefined }
+      it('should return a Grommet Video component', function () {
+        wrapper.instance().renderMedia(videoPropsMock)
+        const returnedValue = renderMediaSpy.returnValues[0]
+        expect(returnedValue.type).to.equal(Video)
+      })
+
+      it('should set the a11yTitle prop with the alt text', function () {
+        wrapper.instance().renderMedia(videoPropsMock)
+        const returnedValue = renderMediaSpy.returnValues[0]
+        expect(returnedValue.props.a11yTitle).to.equal(videoPropsMock.alt)
+      })
+
+      it('should set the src attribute', function () {
+        wrapper.instance().renderMedia(videoPropsMock)
+        const returnedValue = renderMediaSpy.returnValues[0]
+        expect(returnedValue.props.src).to.equal(videoPropsMock.src)
+      })
+    })
+
+    describe('when the media is a video', function () {
+      const audioPropsMock = { src: 'https://panoptes-uploads.zooniverse.org/production/subject_location/1c93591f-5d7e-4129-a6da-a65419b88048.mpga', alt: 'Street noise', children: undefined }
```

Comment:
How is `alt` handled for audio elements? It's not tested here.

Reply:
It is not. I think the original markdown-it plugin we were using wasn't setting it since I took a look at that for reference. I could set the alt anyway for audio.

Reply:
I think Grommet sets it as `aria-label` on the video element, so we could copy that behaviour for audio.